### PR TITLE
JavaScript Translations

### PR DESCRIPTION
**Pull request by biyeun**
@dimagi/js-team Added ability to translate javascript files. In order for django to pick up on changes, 
`./manage.py makemessages -d djangojs --all` will need to be run in addition to the standard `makemessages` command. `compilemessages` should be run as normal.

In addition to this, you'll need to run `./manage.py compilejsi18n` to make the translation utilities available in all js files.

### CODE REVIEW

File: corehq/apps/app_manager/static/app_manager/js/commcaresettings.js [hunk 1]
```diff
@@ -159,7 +161,7 @@ function CommcareSettings(options) {
                     uniqueNames = names.filter(function(elem, pos) {
                         return names.indexOf(elem) == pos;
                     })
-                    return 'Auto-set by ' + uniqueNames.join(', ')
+                    return gettext('Auto-set by ') + uniqueNames.join(', ')
```
Comment (dimagimon) [inline corehq/apps/app_manager/static/app_manager/js/commcaresettings.js]: Missing semicolon.

**Comment by biyeun** [comment]
@dimagi/team-commcare-hq when this is merged, you'll need to run `manage.py compilejsi18n`

File: corehq/apps/app_manager/static/app_manager/js/commcaresettings.js (hunk 1 above)
Comment (dimagimon) [inline corehq/apps/app_manager/static/app_manager/js/commcaresettings.js]: Missing semicolon.

**Comment by proteusvacuum** [comment]
cool! does this work the same way that `ugettext` works? mainly wondering what the failure mode is if a translation doesn't exist in a specific language (I've had issues with js i18n libraries in the past that would spit a bunch of junk into my template if it couldn't find the right translation).

**Comment by benrudolph** [comment]
amazing!! :heart_eyes: much needed. is there a way we can smartly pull down just the translations that we need? or will we have to pull down the entire file each time? i imagine that those files could be pretty hefty over time

**Comment by czue** [comment]
awesome!! :fr: :es: :it:

**Comment by dannyroberts** [comment]
Awesome, @biyeun!! This is a huge piece in giving us the freedom to decouple our templates and JS however we want, instead of being forced to put some things in django templates just to be able to use translations! :star2:

**Comment by orangejenny** [comment]
this is great! also, looks good to merge.

**Comment by snopoke** [comment]
Was there a reason you chose to use this lib vs django's builtin support: https://docs.djangoproject.com/en/1.7/topics/i18n/translation/#internationalization-in-javascript-code

**Comment by biyeun** [comment]
@snopoke this library has a management command which generates the js translation file and places it in the staticfiles directory. Django's built in support is still there for the translation part, but the way it serves the file is done on a per request basis, unless you cache the request.

**Comment by biyeun** [comment]
@proteusvacuum it does work the same as ugettext https://docs.djangoproject.com/en/1.7/topics/i18n/translation/#internationalization-in-javascript-code

**Comment by millerdev** [comment]
sweetness!

**Comment by biyeun** [comment]
@benrudolph 

> is there a way we can smartly pull down just the translations that we need? or will we have to pull down the entire file each time? i imagine that those files could be pretty hefty over time

I think there is a way to, but we'd have to spin off a separate thing from `statici18n`. Also, once the file is loaded, the browser should cache it anyway.

**Comment by benrudolph** [comment]
gotcha makes sense. thanks

**Comment by benrudolph** [comment]
:earth_asia: :earth_africa: :earth_americas:

**Comment by dannyroberts** [comment]
`_("Awesome")`!

**Comment by czue** [comment]
:airplane: :airplane: :airplane:

**Pull request merged**